commons/src/main/java/org/dhis2/commons/resources/LocaleSelector.kt

```diff
@@ -43,35 +44,39 @@ class LocaleSelector(private val base: Context, private val d2: D2) {
             .blockingGet()?.value()
     }
 
-    private fun hasLanguageChanged(): Boolean {
-        val userLanguage = getUserLanguage()
-        val currentLanguage = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            configuration.locales[0].toLanguageTag()
-        } else {
-            configuration.locale.language
-        }
-        return userLanguage != null && userLanguage != currentLanguage
-    }
-
     private fun changeToUserLanguage(base: Context): Context {
```

Comment:
Would be possible to test this given that you have injected D2 and the context?.